lib/Gitlab/Api/Repositories.php

```diff
@@ -4,11 +4,15 @@ class Repositories extends AbstractApi
 {
     /**
      * @param int $project_id
+     * @param array $parameters
      * @return mixed
      */
-    public function branches($project_id)
+    public function branches($project_id, array $parameters = [])
     {
-        return $this->get($this->getProjectPath($project_id, 'repository/branches'));
+        $resolver = $this->createOptionsResolver();
+        $resolver->setDefined('query');
```

Comment:
According to https://docs.gitlab.com/ee/api/branches.html#list-repository-branches, there is no `query` filter available for this branch. Could you remove it?

Reply:
As the filter is undocumented, I would not add it for now and try to learn more about why it's not documented.

Reply:
Without this fix you can't load all branches.

Reply:
You can using the `Gitlab\ResultPager::fetchAll()`!

However, I'm not willing to add undocumented methods/parameters to this library. There is probably a reason about it not being documented. We can move forward with this PR by removing the `query` option, and add-it later if it was a Gitlab API documentation issue.